iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,127 +37,206 @@ class Duration;
     template <typename T>
-    static constexpr Duration milliseconds(const T ms);
+    static constexpr Duration milliseconds(const T value);
+
+    /// @brief Constructs a new Duration object from seconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as seconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration seconds(const T seconds);
+    static constexpr Duration seconds(const T value);
+
+    /// @brief Constructs a new Duration object from minutes
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as minutes
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration minutes(const T min);
+    static constexpr Duration minutes(const T value);
+
+    /// @brief Constructs a new Duration object from hours
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as hours
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration hours(const T hours);
+    static constexpr Duration hours(const T value);
+
+    /// @brief Constructs a new Duration object from days
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as days
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration days(const T days);
+    static constexpr Duration days(const T value);
+
+    // END CREATION FROM STATIC FUNCTIONS
+
+    // BEGIN CONSTRUCTORS AND ASSIGNMENT
+
+    /// @brief Constructs a Duration from seconds and nanoseconds
+    /// @param[in] seconds portion of the duration
+    /// @param[in] nanoseconds portion of the duration
+    constexpr Duration(const uint64_t seconds, const uint32_t nanoseconds);
 
     /// @brief Construct a Duration object from timeval
+    /// @param[in] value as timeval
     constexpr explicit Duration(const struct timeval& value);
 
     /// @brief Construct a Duration object from timespec
+    /// @param[in] value as timespec
     constexpr explicit Duration(const struct timespec& value);
 
     /// @brief Construct a Duration object from itimerspec
+    /// @param[in] value as itimespec
+    /// @note only it_interval from the itimerspec is used
     constexpr explicit Duration(const struct itimerspec& value);
 
     /// @brief Construct a Duration object from std::chrono::milliseconds
+    /// @param[in] value as milliseconds
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     constexpr explicit Duration(const std::chrono::milliseconds& value);
```

Comment:
In a follow up pull request we should templatize this somehow so that all `std::chrono::` timetypes are allowed.